tools/kliopt/org/jetbrains/kliopt/KliOpt.kt

```diff
@@ -17,18 +17,23 @@
 
 package org.jetbrains.kliopt
 
-enum class DeprecatedState {
-    ALL {
-        override fun helpMessage(descriptor: Descriptor): String = "Warning: Option/argument -${descriptor.longName} is deprecated!"
-    },
-    ONLY_SHORT {
-        override fun helpMessage(descriptor: Descriptor): String =
-                "Warning: Short form -${(descriptor as? OptionDescriptor)?.shortName} of option -${descriptor.longName} is deprecated!"
-    },
-    ONLY_FULL {
-        override fun helpMessage(descriptor: Descriptor): String =
-                "Warning: Full form of option -${descriptor.longName} is deprecated!"
-    };
+sealed class DeprecatedState(val newName: String? = null) {
```

Comment:
The entire machinery looks much like a case of over-engineering to me.
Most of the features aren't used currently. Do we have any examples except for `-h`?

Reply:
It's common tool, it can be used not only in interop tools.